options.js

```diff
@@ -163,27 +166,65 @@ function createKeyBindings(item) {
   keyBindings.push({action: action, key: key, value: value, force: force, predefined: predefined});
 }
 
+// Validates settings before saving
+function validate() {
+  var valid = true;
+  var status = document.getElementById('status');
+  document.getElementById('blacklist').value.split("\n").forEach(match => {
+    match = match.replace(regStrip,'')
+    if (match.startsWith('/')) {
+      try {
+        var regexp = new RegExp(match);
+      } catch(err) {
+        status.textContent = 'Error: Invalid blacklist regex: ' + match
+                           + '. Unable to save';
+        valid = false;
+        return;
+      }
+    }
+  })
+  document.getElementById('blacklistrc').value.split("\n").forEach(match => {
+    match = match.replace(regStrip,'')
+    if (match.startsWith('/')) {
+      try {
+        var regexp = new RegExp(match);
+      } catch(err) {
+        status.textContent = 'Error: Invalid ratechange blacklist regex: ' + match
+                           + '. Unable to save';
+        valid = false;
+        return;
+      }
+    }
+  })
+  return valid;
+}
+
 // Saves options to chrome.storage
 function save_options() {
+  if (validate() === false) {
+    return;
+  }
   keyBindings = [];
   Array.from(document.querySelectorAll(".customs")).forEach(item => createKeyBindings(item)); // Remove added shortcuts
 
-  var rememberSpeed = document.getElementById('rememberSpeed').checked;
-  var audioBoolean = document.getElementById('audioBoolean').checked;
-  var enabled = document.getElementById('enabled').checked;
-  var startHidden = document.getElementById('startHidden').checked;
+  var rememberSpeed     = document.getElementById('rememberSpeed').checked;
+  var audioBoolean      = document.getElementById('audioBoolean').checked;
+  var enabled           = document.getElementById('enabled').checked;
+  var startHidden       = document.getElementById('startHidden').checked;
   var controllerOpacity = document.getElementById('controllerOpacity').value;
-  var blacklist     = document.getElementById('blacklist').value;
+  var blacklist         = document.getElementById('blacklist').value;
+  var blacklistrc       = document.getElementById('blacklistrc').value;
 
   chrome.storage.sync.remove(["resetSpeed", "speedStep", "fastSpeed", "rewindTime", "advanceTime", "resetKeyCode", "slowerKeyCode", "fasterKeyCode", "rewindKeyCode", "advanceKeyCode", "fastKeyCode"]);
   chrome.storage.sync.set({
-    rememberSpeed:  rememberSpeed,
-    audioBoolean:  audioBoolean,
-    enabled:  enabled,
```

Comment:
This section was a confusing mess that seemed to follow no style guide... so i took some style liberties that seemed to make sense in this context.